src/aggregate/aggregate_exec.c

```diff
@@ -292,6 +293,8 @@ void sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
   req->qiter.totalResults = 0;
   if (resultsLen == REDISMODULE_POSTPONED_ARRAY_LEN) {
     RedisModule_ReplySetArrayLength(outctx, nelem);
+  } else {
+    RS_LOG_ASSERT(resultsLen == nelem, "Precalculated number of replies must be equal to actual number");
```

Comment:
The problem here is that we don't have an effective way of returning an error and aborting the transaction gracefully (rather than stopping the world). Also, when aborting gracefully, we need to make sure we log the error.
I'd suggest we look around to see if an error can be enabled, and if it's not easy to do, we'll wait for exceptions to be around.

Reply:
The connection to the client is broken beyond repair when we mess the array length.